operator/src/main/resources/scripts/introspectDomain.py

```diff
@@ -869,6 +879,54 @@ def customizeLog(self, name, bean, isDomainBean):
     self.undent()
     self.writeln("</d:log>")
 
+  def customizeFileStores(self, domain):
+    data_dir = self.env.getDataHome()
+    if data_dir is None or len(data_dir) == 0:
+      # do not override if dataHome not specified or empty ("")
+      return
+
+    for filestore in domain.getFileStores():
+      self.customizeFileStore(filestore, data_dir)
+
+
+  def customizeFileStore(self, filestore, data_dir):
+    fileaction=''
+    if filestore.getDirectory() is None:
+      fileaction=' f:combine-mode="add"'
+    else:
+      fileaction=' f:combine-mode="replace"'
+
+    self.writeln("<d:file-store>")
+    self.indent()
+    self.writeln("<d:name>" + filestore.getName() + "</d:name>")
+    self.writeln("<d:directory"+ fileaction + ">" + data_dir + "</d:directory>")
+    self.undent()
+    self.writeln("</d:file-store>")
+
+  def customizeDefaultFileStore(self, bean):
+    data_dir = self.env.getDataHome()
+    if data_dir is None or len(data_dir) == 0:
+      # do not override if dataHome not specified or empty ("")
+      return
+
+    dfsaction=''
+    fileaction=''
+    if bean.getDefaultFileStore() is None:
+      # don't know why, but don't need to "add" a missing domain log bean, and adding it causes trouble
+      dfsaction=' f:combine-mode="add"'
+      fileaction=' f:combine-mode="add"'
```

Comment:
1 - Comment is incorrect - mentions log bean.
2 - This combination needs to be tested (overriding data dir when default store bean is missing).

Reply:
Tom, can you point me to any test framework/code we have for creating a WLS Domain and running multiple scenarios in order to test the different configurations above by the instrospector?  I've looked at src/integration-tests/introspector but it's unclear if it can be run multiple individual tests or it just runs  a full suite of tests based on one configuration.

Reply:
I'd be tempted to just try this out 'ad hoc', and the write-up a test plan JIRA for the QA team to create formal tests.You can overload the existing test in src/integration-tests/introspector  - just add more config as needed to the domain create (wl-create-domain-pod.pyt), and then, at the end, verify (a) that overrides are reflected in the bean tree, and (b)

Reply:
To 'ad hoc' test one way might be temporarily modify introspector integration tests:
Modify domain config (wl-create-domain-pod.pyt) 'admin server' to have one style of default store config, 'cluster' to have a second style, plus add two custom file stores on admin server, and something that puts data in the custom file stores  to force them to create .DAT files (JMS Server + persistent messages?).   Then (a) verify overrides are taking effects via WLST inspect of the bean tree, and (b) verify overrides taking effect by noting the .DAT files are in the expected location.  Maybe just do (b) as (b) implies (a) must be working.

Reply:
Thank you for the guidance, I will look into how to implement your 'ad hoc' tests.
Also, QA has already filed a JIRA: https://jira.oraclecorp.com/jira/browse/OWLS-75000 which we can update with your test coverage scenarios.